docs/guides/using-mtls-auth-plugin.md

```diff
@@ -0,0 +1,291 @@
+Kong for Kubernetes Enterprise on your Kubernetes cluster.
+
+## Testing Connectivity to Kong
+
+This guide assumes that the `PROXY_IP` environment variable is
+set to contain the IP address or URL pointing to Kong.
+Please follow one of the
+[deployment guides](../deployment/k4k8s-enterprise.md) to configure
+this environment variable.
+
+If everything is setup correctly, making a request to Kong should return
+HTTP 404 Not Found.
+
+```bash
+$ curl -i $PROXY_IP
+HTTP/1.1 404 Not Found
+Date: Fri, 21 Jun 2019 17:01:07 GMT
+Content-Type: application/json; charset=utf-8
+Connection: keep-alive
+Content-Length: 48
+Server: kong/1.2.1
+
+{"message":"no Route matched with those values"}
+```
+
+This is expected as Kong does not yet know how to proxy the request.
+
+## Provision a CA certificate in Kong
+
+CA certificates in Kong are provisioned by create a `Secret` resource in
+Kubernetes.
+
+The secret resource should have a few properties:
+- It should have the `konghq.com/ca-cert: "true"` label.
+- It should have a `cert` data property which contains a valid CA certificate
+  in PEM format.
+- It should have an `id` data property which contains a random UUID.
+
+Note that a self-signed CA certificate is being used for the purpose of this
+guide. You should use your own CA certificate that is backed by
+your PKI infrastructure.
+
+```bash
+$ echo "apiVersion: v1
+kind: Secret
+metadata:
+  name: my-ca-cert
+  labels:
+    konghq.com/ca-cert: 'true'
+type: Opaque
+stringData:
+  cert: |
+    -----BEGIN CERTIFICATE-----
+    MIICwTCCAamgAwIBAgIUHGUzUWvHJHrREvIZIcORiFUvze4wDQYJKoZIhvcNAQEL
+    BQAwEDEOMAwGA1UEAwwFSGVsbG8wHhcNMjAwNTA4MjExODA1WhcNMjAwNjA3MjEx
+    ODA1WjAQMQ4wDAYDVQQDDAVIZWxsbzCCASIwDQYJKoZIhvcNAQEBBQADggEPADCC
+    AQoCggEBANCMMBngjuTvqts8ZXtZhqdr181QH/NmytW1KlyqZd6ppXUer+i0OWhP
+    1nAyHsBPJljKAFLd8l1EioPFkN78/wJFDJrHOtfniIQPVLdS2cnNQ72dLyQH6smH
+    JQDV8ePBQ2GdRP6s61+Da8eoaW6nSLtmEUhxvyteboqwmi2CtUtAfuiU1m5sOdpS
+    z+L4D08CE+SFIT4MGD3gxNdg7lccWCHIfk54VRSdGDKEVwed8OQvxD0TdpHY+ym5
+    nJ4JSkhiS9XIodnxR3AZ6rIPRqk+MQ4LGTjX2EbM0/Yg4qvnZ7m4fcpK2goDZIVL
+    EF8F+ka1RaAYWTsXI1BAkJbb3kdo/yUCAwEAAaMTMBEwDwYDVR0TBAgwBgEB/wIB
+    ADANBgkqhkiG9w0BAQsFAAOCAQEAVvB/PeVZpeQ7q2IQQQpADtTd8+22Ma3jNZQD
+    EkWGZEQLkRws4EJNCCIvkApzpx1GqRcLLL9lbV+iCSiIdlR5W9HtK07VZ318gpsG
+    aTMNrP9/2XWTBzdHWaeZKmRKB04H4z7V2Dl58D+wxjdqNWsMIHeqqPNKGamk/q8k
+    YFNqNwisRxMhU6qPOpOj5Swl2jLTuVMAeGWBWmPGU2MUoaJb8sc2Vix9KXcyDZIr
+    eidkzkqSrjNzI0yJ2gdCDRS4/Rw9iV3B3SRMs0mJMLBDrsowhNfLAd8I3NHzLwps
+    dZFcvZcT/p717K3hlFVdjGnKIgKcG7aYji/XRR87HKnc+cJMCw==
+    -----END CERTIFICATE-----
+  id: cce8c384-721f-4f58-85dd-50834e3e733a" | kubectl create -f -
+secret/my-ca-cert created
+```
+
+Please note the ID, you can use this ID one or use a different one but
+the ID is important in the next step when we create the plugin.
+Each CA certificate that you create needs a unique ID.
+Any random UUID will suffice here and it doesn't have an security
+implication.
```

Comment:
Suggest adding an example:

You can use [uuidgen](https://linux.die.net/man/1/uuidgen) (Linux, OS X) or [New-Guid](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/new-guid) (Windows) to generate an ID, e.g.
```
$ uuidgen 
05f5fc12-3f6a-4e44-bcd2-a9780e73029d
```

Reply:
TIL